lib/utils/dataflow/src/index.js

```diff
@@ -335,11 +358,28 @@ const logOutput = function *(odoc, odb) {
 };
 
 // Log a list of output docs
-const logOutputs = function *(odocs, odb) {
-  yield tmap(odocs.concat([]).reverse(), function *(odoc, i, l) {
-    // Log each doc into the output database
-    yield logOutput(odoc, odb);
-  });
+const logOutputs = function *(odocs, odb, edb) {
+  let errors;
+  // if there is an error, post to error db
+  if(!validateDocs(odocs).length) {
```

Comment:
Is this really a validation? Looks like it's only checking for the presence of an error in any doc, and then either returning an empty array or the whole list of docs, so that doesn't look like a validation to me, and also not sure about the purpose of returning an array (empty or the input array) instead of just a true/false.